hax/hax/types.py

```diff
@@ -216,3 +216,41 @@ def __repr__(self):
 
 
 HAState = NamedTuple('HAState', [('fid', Fid), ('status', ServiceHealth)])
+
+
+class m0HaProcessEvent(Enum):
+    M0_CONF_HA_PROCESS_STARTING = 0
+    M0_CONF_HA_PROCESS_STARTED = 1
+    M0_CONF_HA_PROCESS_STOPPING = 2
+    M0_CONF_HA_PROCESS_STOPPED = 3
+
+    def str_to_Enum(self):
+        events = {'M0_CONF_HA_PROCESS_STARTING':
+                  self.M0_CONF_HA_PROCESS_STARTING,
+                  'M0_CONF_HA_PROCESS_STARTED':
+                  self.M0_CONF_HA_PROCESS_STARTED,
+                  'M0_CONF_HA_PROCESS_STOPPING':
+                  self.M0_CONF_HA_PROCESS_STOPPING,
+                  'M0_CONF_HA_PROCESS_STOPPED':
+                  self.M0_CONF_HA_PROCESS_STOPPED}
+        return events[self.name]
+
+    def __repr__(self):
+        return self.name
+
+
+class m0HaProcessType(Enum):
+    M0_CONF_HA_PROCESS_OTHER = 0
+    M0_CONF_HA_PROCESS_KERNEL = 1
+    M0_CONF_HA_PROCESS_M0MKFS = 2
+    M0_CONF_HA_PROCESS_M0D = 3
+
+    def str_to_Enum(self):
```

Comment:
Do you need this method? I didn't find in the sources you use it.

And anyway, it seems like you don't need it (at least, it could have been way simpler: `m0HaProcessType['<your string here>']`:
```python
>>> class testme(Enum):
...     VAL1 = 0
...     VAL2 = 1
...
>>> t =testme(1)
>>> t
<testme.VAL2: 1>
>>> t =testme('VAL1')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib64/python3.6/enum.py", line 293, in __call__
    return cls.__new__(cls, value)
  File "/usr/lib64/python3.6/enum.py", line 535, in __new__
    return cls._missing_(value)
  File "/usr/lib64/python3.6/enum.py", line 548, in _missing_
    raise ValueError("%r is not a valid %s" % (value, cls.__name__))
ValueError: 'VAL1' is not a valid testme
>>> t = testme['VAL1']
>>> t
<testme.VAL1: 0>
```

Reply:
Reverted changes.